src/ReceiveQueueListener.java

```diff
@@ -0,0 +1,121 @@
+
+package networking;
+import java.util.HashMap;
+import networking.queueManagement.*;
+import networking.utility.*;
+import infrastructure.validation.logger.*;
+/**
+ * 
+ * The class "ReceiveQueueListener" helps in dequeuing the receiving queue and calling handler method
+ * of the corresponding module ( either processing module or content module) by passing the message.
+ * 
+ * @author Drisya P
+ * @since 2021-11-31
+ */
+public class ReceiveQueueListener implements Runnable{
+    
+    /**
+     * Receiving queue of the processing module or content module based.
+     */
+    private IQueue<IncomingPacket> receivequeue;
+    
+    /**
+     * Logger instance for logging errors and activities.
+     */
+    ILogger logger;
+    
+    /**
+     * Map stores module identifiers ( processing / content module ) and corresponding handler. 
+     */
+    private HashMap<String, INotificationHandler> handlerMap;
+    
+    /**
+     * Creates a new ReceiveQueueListener with the given queue and handlerMap.
+     * @param receivequeue     processing or content module's receiving queue.
+     * @param handlerMap       map which contains identifier and corresponding module INotificationHandler handler.
+    */
+    public ReceiveQueueListener( IQueue<IncomingPacket> receivequeue, HashMap<String, INotificationHandler> handlerMap ) {
+        
+        this. receivequeue = receivequeue;
+        this.handlerMap = handlerMap;
+        
+    }
+    
+    
+    /**
+     * The method "run" is used to dequeue from the queue continuously when the queue is non empty and call
+     * helper function to send the message to corresponding module.
+     * @return Nothing
+     * @exception exception    Error when unable to call onCallback method.
+     * @see Exception
+     */
+
+    @Override
+    public void run() {
+       
+        
+        while(true){
```

Comment:
Use LanCommunicator.isRunning for while condition to make it consistent with other listeners.

Reply:
Thanks you @sunny117  for pointing it out

Reply:
@Drisya-Ponmari  please update this

Reply:
@Drisya-Ponmari  and also please keep this file in the folder src/networking/

Reply:
> Use LanCommunicator.isRunning for while condition to make it consistent with other listeners.

Yeah correct. Thanks

Reply:
> @Drisya-Ponmari and also please keep this file in the folder src/networking/

Okay

Reply:
For a change use Lancommunicator.getStatus( ) for isRunning